packages/ui-extensions/docs/surfaces/admin/create-doc-files.sh

```diff
@@ -1,84 +1,82 @@
 #!/bin/bash
 
-# Validate surface argument
-surface="$1"
+# Check if at least one component name is provided
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 [-t type] component_name1 [component_name2 ...]"
+    echo "  -t: Type of documentation (api or components, default: components)"
+    exit 1
+fi
 
-echo "surface: $surface"
+# Parse arguments
+type="components"  # default value
+while getopts "t:" opt; do
+    case $opt in
+        t) type="$OPTARG"
+           if [[ "$type" != "components" && "$type" != "api" ]]; then
+               echo "Error: type must be either 'components' or 'api'"
+               exit 1
+           fi
+           ;;
+        \?) echo "Invalid option -$OPTARG"
+            exit 1
+            ;;
+    esac
+done
+shift $((OPTIND-1))
 
-if [ "$surface" = "admin" ] || [ "$surface" = "checkout" ]; then
-  echo "Scaffolding docs for ${surface}"
-else
-  echo "ERR: No surface argument provided. Please provide a surface argument (admin or checkout)."
-  exit 1
-fi
 
-# Make this an input to the script
-components_dir="src/surfaces/${surface}/components"
-# components_dir="./packages/ui-extensions/src/surfaces/${surface}/components"
+echo "Scaffolding ${type} docs for admin"
 
-# components_dir="./packages/ui-extensions/src/surfaces/${surface}/api"
+base_dir="./packages/ui-extensions/src/surfaces/admin/${type}"
 
-addExamples() {
-    folder=$1
-    if [ -d "${folder}/examples" ]; then
-      example_folder_path="${folder}/examples"
-      examples=("$example_folder_path"/*)
+for componentName in "$@"; do
+    # Create component directory
+    folder="${base_dir}/${componentName}"
+    mkdir -p "$folder"
 
-      if [ ${#examples[@]} -eq 1 ] && [[ ${examples[0]} == *"basic"* ]]; then
-        echo ""
-        return
-      fi
+    examples_dir="${folder}/examples"
+    mkdir -p "$examples_dir"
+    
+    docs_file="${folder}/${componentName}.doc.ts"
+    lowercaseComponentName=$(echo "$componentName" | tr '[:upper:]' '[:lower:]')
```

Comment:
Can we turn these into lowercase dashcasing?

Reply:
Fixed 🙂